src/client/pluginChannels.js

```diff
@@ -93,15 +93,21 @@ module.exports = function (client, options) {
     return ret
   }
 
-  function writeDumbArr (value, buf, offset) {
-    // TODO: Remove trailing \0
-    value.forEach(function (v) {
-      offset += proto.write(v, buf, offset, 'cstring')
+  function writeDumbArr (arr, buf, offset) {
+    arr.forEach(function (v, k) {
+      offset += proto.write(v, buf, offset, ['pstring', []])
+      if (k < arr.length - 1) {
+        buf.writeInt8(0x00, offset)
```

Comment:
Why are you writing this here?

Reply:
Writing what exactly? `minecraft:register` expects contents to be separated by null (see [wiki.vg](https://wiki.vg/Plugin_channels#minecraft:register)).

Reply:
Then what this should be is an array of c strings which are null terminated (like the current code). I don't see the need for the change over the existing implementation. If there is no null terminator at the end it can be sliced from the buffer, or something like buffer.write(value.join(\0))